services/context/repo.go

```diff
@@ -84,7 +84,18 @@ func (r *Repository) CanCreateBranch() bool {
 }
 
 func (r *Repository) GetObjectFormat() git.ObjectFormat {
-	return git.ObjectFormatFromName(r.Repository.ObjectFormatName)
+	if r.Repository.ObjectFormatName != "" {
+		return git.ObjectFormatFromName(r.Repository.ObjectFormatName)
+	}
+	if r.GitRepo != nil {
+		objectFormat, err := r.GitRepo.GetObjectFormat()
+		if err == nil {
+			return objectFormat
+		}
+		log.Error("Failed to get object format: %v", err)
+	}
+
+	return git.Sha1ObjectFormat
```

Comment:
This logic couldn't be right.

Reply:
f887ed08074cc7b7723b8ba0a85c69835cc1bddd